### ISSUE TITLE
db/migration/V4.14.0.1__5822-export-var-meta.sql deployment failures on develop

### ISSUE THREAD

**Issue by donsizemore** [IQSS/dataverse#5862]
`[2019-05-20T18:56:39.496+0000] [glassfish 4.1] [SEVERE] [] [javax.enterprise.system.tools.deployment.common] [tid: _ThreadID=46 _ThreadName=admin-listener(5)] [timeMillis: 1558378599496] [levelValue: 1000] [[
  Exception while invoking class org.glassfish.ejb.startup.EjbApplication start method
javax.ejb.EJBException: javax.ejb.CreateException: Initialization failed for Singleton StartupFlywayMigrator`

`Caused by: org.flywaydb.core.internal.command.DbMigrate$FlywayMigrateException: 
Migration V4.14.0.1__5822-export-var-meta.sql failed
SQL State  : 42701
Error Code : 0
Message    : ERROR: column "postquestion" of relation "variablemetadata" already exists
Location   : db/migration/V4.14.0.1__5822-export-var-meta.sql`

`Caused by: org.flywaydb.core.internal.sqlscript.FlywaySqlScriptException: 
Migration V4.14.0.1__5822-export-var-meta.sql failed
SQL State  : 42701
Error Code : 0
Message    : ERROR: column "postquestion" of relation "variablemetadata" already exists
Location   : db/migration/V4.14.0.1__5822-export-var-meta.sql`

etc. observed via EC2-create/dataverse-ansible/phoenix

**Comment by pdurbin** [comment]
As I mentioned at http://irclog.iq.harvard.edu/dataverse/2019-05-20#i_94408 I'm seeing the same problems on the "phoenix" server. The deployment phase does not succeed so the API test phase never runs.

https://build.hmdc.harvard.edu:8443/job/phoenix.dataverse.org-deploy-develop/407/console is an example failure and server.log shows similar lines that @donsizemore is seeing:

`Message    : ERROR: column "postquestion" of relation "variablemetadata" already exists`

`Location   : db/migration/V4.14.0.1__5822-export-var-meta.sql (/usr/local/glassfish4/glassfish/domains/doma​in1/applications/dataverse/WEB-INF/classes/db​/migration/V4.14.0.1__5822-export-var-meta.sql)`

This Flyway script was added in pull request #5833 and I'm wondering... is this the first time we've added a column using Flyway? From a quick look it seems like it is. We must be doing it wrong.

Right now we're doing this:

`ALTER TABLE variablemetadata ADD COLUMN postquestion text;`

Maybe we need to add some sort of "if doesn't exist" to `V4.14.0.1__5822-export-var-meta.sql`?

**Comment by poikilotherm** [comment]
Just a quick shot: this is a new deployment, right? Has the Postgres DB been initialized from scratch? Then the persistence framework might have setup all fields already, causing migration to fail.

See also: https://github.com/IQSS/dataverse/blob/aa59755ded639af2f33a22d52f6b7cfca33a7db4/src/main/resources/META-INF/persistence.xml#L14

From what I read on articles about Flyway + some kind of persistence framework: don't let the framework manage DDL generation for you. Maybe @MrK191 can shed some light on this, too?

**Comment by pdurbin** [comment]
According to https://stackoverflow.com/questions/12597465/how-to-add-column-if-not-exists-on-postgresql/38721951#38721951 as of PostgreSQL 9.6 we can use syntax like this:

    ALTER TABLE table_name ADD COLUMN IF NOT EXISTS column_name INTEGER;

So, for us I guess that would be

    ALTER TABLE variablemetadata ADD COLUMN IF NOT EXISTS postquestion text;

**Comment by MrK191** [comment]
@poikilotherm Yes exactly,  <property name="eclipselink.ddl-generation" value="create-tables"/>  needs to be value="none".

**Comment by poikilotherm** [comment]
As requested by @pdurbin [on IRC today](http://irclog.iq.harvard.edu/dataverse/2019-05-21#i_94465), I'll open another issue for this.

#5833  contains a quick fix/workaround, but might get refactored depending on the results from the new issue.

**Comment by pdurbin** [comment]
As discussed in http://irclog.iq.harvard.edu/dataverse/2019-05-21 a pull request with the quick fix suggested above doesn't exist yet but @donsizemore is going to make one. Thanks!!

@poikilotherm @MrK191 thanks for your feedback and please post the new issue here when you're ready. Please note that in pull request #5858 we recently decided to stop making the "create" scripts, the `dataverse_VDCNet-ejbPU_createDDL.jdbc` file but if these create scripts are needed for what you're envisioning in the new issue, you might want to spell this out. Thanks!!

**Comment by djbrooke** [comment]
Big thanks @MrK191 @donsizemore @poikilotherm and @pdurbin for the troubleshooting and the fix!

**Comment by pdurbin** [comment]
Pull request #5866 looks like exactly what I was suggesting above as a quick fix. Thanks, @donsizemore . I hope it works.

To test: try to install Dataverse fresh from the develop branch. It should fail with the errors above. Then try the pull request above. Hopefully it works.

**Comment by donsizemore** [comment]
@pdurbin I tested it in EC2 against my fork and branch; I encourage checking behind me!

**Comment by pdurbin** [comment]
> From what I read on articles about Flyway + some kind of persistence framework: don't let the framework manage DDL generation for you.

Here's the follow up issue:  Flyway: disable DDL generation from EclipseLink #5871

**Comment by pdurbin** [comment]
@donsizemore the phoenix server needs a postgre upgrade from 9.3 to 9.6 (#5872) but are you unblocked? I assume so because your dataverse-ansible installs postgres 9.6.
